doc/pmreorder/pmreorder.1.md

```diff
@@ -0,0 +1,242 @@
+a persistent memory checking tool.
+
+Pmreorder performs store reordering between prersistent
+memory barriers - a sequence of flush-fence operations.
+It uses the consistency checking routine provided in the
+command line options to check whether the files are in a
+consistent state.
+
+Seeing as the combination of valgrind and the multiple store
+reordering will take a lot of time to process, it would be
+beneficial to do as little stores as possible to verify
+whether the tested program is powerfail safe.
+
+
+# OPTIONS #
+
+`-h, --help`
+Prints synopsis and list of options.
+
+`-l <store_log>, --logfile <store_log>`
+The pmemcheck log file to process.
+
+`-c <prog|lib>, --checker <prog|lib>`
+Consistency checker type.
+
+`-p <path>, --path <path>`
+Path to the consistency checker.
+
+`-n <name>, --name <name>`
+The name of the consistency checking function
+within the library.
+
+`-t <print|file>, --output_type <print|file>`
+Set the logger type. Default value is `print`.
+
+`-o <pmreorder_output>, --output <pmreorder_output>`
+Set the logger output file.
+(If `-t` option is set to `file`).
+
+`-e <debug|info|warning|error|critical>,
+ --output_level <debug|info|warning|error|critical>`
+Set the output log level.
+
+`-r  <full|noreorder|accumulative|partial>,
+ --default_engine  <full|noreorder|accumulative|partial>`
+Set the default reorder engine. Default value is `full`.
+
+
+# ENGINES #
+
+By default all possible permutations of stores are tested.
+
+In such case to check consistency of stores between barriers
+pmreorder uses `full` reorder engine. It means that for each
+set of stores it generates all possible permutations to check.
+This may takes a significant amount of time, so there is avaiable
+a few more types of engines to limit number of tested sequences:
+
++ **noreorder** - pass set of stores unchanged, do check
+on the original collection. It can be useful for less
+significant parts of code.
+
+```
+Example:
+        input: (a, b, c)
+        output: (a, b, c)
+```
+
++ **accumulative** - perform check on growing sequences
+
+```
+Example:
+        input: (a, b, c)
+        output:
+               ()
+               ('a')
+               ('a', 'b')
+               ('a', 'b', 'c')
+```
+
++ **partial** - generates a sequence of all possible combinations
+without replacement of stores (limited to 1000), and then select
+from them random 3 sequences to run under consistency checker.
+
+```
+ Example:
+         input: (a, b, c)
+         output:
+                ('b', 'c')
+                ('b',)
+                ('a', 'b', 'c')
+```
+
+
+# INSTRUMENTALIZATION #
+
+Another way to switch types of reordering is to use Valgrind-pmemcheck
+instrumentalization at runtime within the tested program.
+
+Exposed macros:
+
++ **VALGRIND_LOG_STORES**
+
+Start logging memory operations.
+
++ **VALGRIND_NO_LOG_STORES**
+
+Stop logging memory operations.
+
++ **VALGRIND_FULL_REORDED**
+
+Issue a *full* reorder log.
```

Comment:
Issues a "full" reorder engine marker.

Or something like that. We don't reorder anything in runtime.